docs/prover/approx/hashing.md

```diff
@@ -117,8 +120,77 @@ contract C {
 }
 ```
 
+### Constants in code vs hashes
 
-## Hashing of Unbounded Data
+A special case in Certora Prover's modeling of hashing is the treatment of 
+constants that appear in the code: The Prover implicitly assumes that the hash 
+function never outputs one of these constants on any of the concrete inputs it 
+gets in that program.
+
+For an example, consider this rule and spec:
+
+```
+// CVL:
+methods {
+    function readAtSlotAddress() external returns (uint) envfree;
+    function updateMap(uint k, uint v) external envfree;
+}
+
+rule foo {
+	uint v1 = readAtSlotAddress();
+
+	uint preImage;
+	uint x; 
+
+	updateMap(preImage, x);
+
+	uint v2 = readAtSlotAddress();
+
+	assert(v1 == v2);
+}
+
+// solidity
+contract C {
+	uint constant slotAddress = 1000000;
+
+	mapping(uint => uint) map;
+
+	function updateMap(uint k, uint v) public {
+		map[k] = v;
+	}
+
+	function readAtSlotAddress() public returns (uint r) {
+		assembly {
+			r := sload(slotAddress)
+		}
+	}
+}
+```
+
+The function `readAtSlotAddress` reads from storage at the slot with the number
+1000000. The function `updateMap` updates a mapping; this means it updates
+storage at a hash computed from its identifier (here 1, since it is the second
+field in the contract) and the key `k`. Now, if `k` is chosen such that
+`keccak(1, k)` equals 1000000, the map update would overwrite that storage
+slot, and thus the assertion in the rule `foo` would be violated.
+
+However, the Certora Prover will return "not violated" for this assertion, since
+it assumes that no hash ever collides with the constant 1000000, which occurs in
+the program.
+
+On the other hand, if we change the contract to leave `slotAddress`
+uninitialized, then Certora Prover will return a violation, since then it can
+choose the values such that `keccak(2, preImage)` == `slotAddress`.
```

Comment:
maybe give a link to a violation that one can see how it looks

Reply:
Huh, you mean to give a link to a run?
Didn't think about doing that in our docs, but yeah, why not -- they just need to be stable, but I guess for the forseeable future they are ..